modules/cudf/src/series.ts

```diff
@@ -875,21 +893,72 @@ export {
 };
 
 function inferType(value: any[]): DataType {
-  if (value.length == 0 || (value.every((val) => typeof val === 'number' || val == null)))
-    return new Float64;
-  if (value.every((val) => typeof val === 'string' || val == null)) return new Utf8String;
-  if (value.every((val) => typeof val === 'bigint' || val == null)) return new Int64;
-  if (value.every((val) => typeof val === 'boolean' || val == null)) return new Bool8;
-  throw new TypeError('Unable to infer type series type, explicit type declaration expected');
+  if (value.length === 0) { return new Float64; }
+  let nullsCount    = 0;
+  let arraysCount   = 0;
+  let objectsCount  = 0;
+  let numbersCount  = 0;
+  let stringsCount  = 0;
+  let bigintsCount  = 0;
+  let booleansCount = 0;
+  let unknownCount  = 0;
+  value.forEach((val) => {
```

Comment:
Much more efficient, since it only loops over the values once!

Reply:
@bryevdv yes I believe it could!